### PR TITLE
CodeAction to inline Tekton Task

### PR DESCRIPTION
**Pull request by evidolob**
Fix: #486 

Demo:

https://user-images.githubusercontent.com/929743/110598024-f4a6b200-8189-11eb-9382-c44ac8ffd4b2.mov

### CODE REVIEW

**Comment by siamaksade** [comment]
Looks great 👏

Would it make sense to make the core action label more clear?  Like “Inline NNNN Task spec”.

cc @vdemeester

**Comment by evidolob** [comment]
>Would it make sense to make the core action label more clear? Like “Inline NNNN Task spec”.

Yes that makes more sense.

**Comment by evidolob** [comment]
New label:
Screenshot 2021-03-10 at 10 48 29

**Comment by vdemeester** [comment]
Nice 👍🏼

**Comment by sudhirverma** [comment]
https://user-images.githubusercontent.com/9924513/110772723-f0f65680-8281-11eb-898a-05fa6cdb5281.mov

when I click on `Inline task spec` it was not able to fetch the steps

**Comment by evidolob** [comment]
@sudhirverma It is not clear for me, after click on inline action, it does nothing?

**Comment by sudhirverma** [comment]
> @sudhirverma It is not clear for me, after click on inline action, it does nothing?

yes, it does nothing for me.

**Comment by evidolob** [comment]
Can you wait for a few seconds? some times it take more time to fetch task yaml.

**Comment by sudhirverma** [comment]
> Can you wait for a few seconds? some times it take more time to fetch task yaml.

I waited for a minute it is not able to fetch task YAML.

**Comment by evidolob** [comment]
@sudhirverma I update progress notification and add error logging, can you try again?

File: src/yaml-support/tkn-code-actions.ts [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    return vscode.window.withProgress({location: vscode.ProgressLocation.Notification, cancellable: false, title: `Loading '${codeAction.taskRefName}' Task...` }, async (): Promise<vscode.CodeAction> => {
+      const uri = tektonFSUri(codeAction.taskKind === TektonYamlType.ClusterTask ? ContextType.CLUSTERTASK : ContextType.TASK , codeAction.taskRefName, 'yaml');
+      try {
+        const taskDoc = await tektonVfsProvider.loadTektonDocument(uri, false);
+        codeAction.edit = new vscode.WorkspaceEdit();
+        codeAction.edit.replace(codeAction.documentUri,
+          new vscode.Range(codeAction.taskRefStartPosition, codeAction.taskRefEndPosition), 
+          this.extractTaskDef(taskDoc, codeAction.taskRefStartPosition.character, codeAction.taskRefEndPosition.character));
+      } catch (err){
+        vscode.window.showErrorMessage('Cannot get Tekton Task definition: ' + err.toString());
```
Comment (sudhirverma) [inline src/yaml-support/tkn-code-actions.ts:63]: can you add telemetry for this feature.

File: src/yaml-support/tkn-code-actions.ts [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+    delete task.metadata.managedFields;
+    delete task.metadata.namespace;
+    delete task.metadata.resourceVersion;
+    delete task.metadata.selfLink;
+    delete task.metadata.uid;
+    delete task.metadata.generation;
+    delete task.metadata.creationTimestamp;
+    delete task.metadata.name;
+    delete task.metadata.ownerReferences;
+    delete task.metadata.annotations['kubectl.kubernetes.io/last-applied-configuration'];
```
Comment (sudhirverma) [inline src/yaml-support/tkn-code-actions.ts:111]: it fails here throw error `Cannot get Tekton Task definition: TypeError: Cannot convert undefined or null to object`

File: src/yaml-support/tkn-code-actions.ts [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+      if (task.metadata.creationTimestamp){
+        delete task.metadata.creationTimestamp;
+      }
+      if (task.metadata.name){
+        delete task.metadata.name;
+      }
+      if (task.metadata.ownerReferences){
+        delete task.metadata.ownerReferences;
+      }
+      if (task.metadata.annotations['kubectl.kubernetes.io/last-applied-configuration']){
```
Comment (sudhirverma) [inline src/yaml-support/tkn-code-actions.ts:134]: now getting `Cannot get Tekton Task definition: TypeError: Cannot read property 'kubectl.kubernetes.io/last-applied-configuration' of undefined` error.

**Review by sudhirverma — APPROVED** [review]

**Pull request merged**